src/Kentico.Xperience.Siteimprove/Services/SiteimproveService.cs

```diff
@@ -0,0 +1,274 @@
+
+            sites = await Get<SiteimproveSites>(requestPath, cancellationToken, client);
+
+            if (sites == null || sites.Sites == null)
+            {
+                return siteID;
+            }
+
+            string domain = GetDomain();
+
+            var site = sites.Sites.FirstOrDefault(s => s.SiteUrl != null && s.SiteUrl.Contains($"://{domain}"));
+
+            if (site == null)
+            {
+                eventLogService.LogError(nameof(SiteimproveService), nameof(GetSiteID), $"Site with URL '{domain}' could not be found on Siteimprove.");
+                return siteID;
+            }
+
+            return site.SiteID;
+        }
+
+
+        internal async Task<bool> IsContentCheckEnabledInternal(CancellationToken? cancellationToken = null)
+        {
+            if (!siteimproveOptions.EnableContentCheck)
+            {
+                return false;
+            }
+
+            var client = HttpClient;
+            bool isContentCheckEnabled = await IsContentCheckReady(client, cancellationToken);
+
+            if (!isContentCheckEnabled && await TryEnableContentCheck(client, cancellationToken))
+            {
+                isContentCheckEnabled = await IsContentCheckReady(client, cancellationToken);
+            }
+
+            return isContentCheckEnabled;
+        }
+
+
+        private string GetDomain()
+        {
+            return RequestContext.CurrentDomain;
+        }
+
+
+        private async Task<bool> TryEnableContentCheck(HttpClient client, CancellationToken? cancellationToken = null)
+        {
+            var response = await Post(SiteimproveConstants.CONTENT_CHECK_PATH, cancellationToken, client);
+
+            if (response == null)
+            {
+                return false;
+            }
+
+            if (!response.IsSuccessStatusCode)
+            {
+                string message = response.StatusCode == HttpStatusCode.PaymentRequired
+                    ? "Cannot enable content check. User not subscribed to Prepublish feature."
+                    : "An error occurred during enabling of content check.";
+
+                eventLogService.LogError(nameof(SiteimproveService), nameof(TryEnableContentCheck), message);
+
+                return false;
+            }
+
+            return true;
+        }
+
+
+        private async Task<bool> IsContentCheckReady(HttpClient client, CancellationToken? cancellationToken = null)
+        {
+            var contentCheck = await Get<SiteimproveContentCheck>(SiteimproveConstants.CONTENT_CHECK_PATH, cancellationToken, client);
+
+            if (contentCheck == null)
+            {
+                return false;
+            }
+
+            return contentCheck.IsReady;
+        }
+
+
+        private async Task<string> GetTokenInternal(CancellationToken? cancellationToken = null)
+        {
+            string token = SettingsKeyInfoProvider.GetValue(SiteimproveConstants.TOKEN_SETTINGS_KEY_NAME);

```

Comment:
There is already a class field with the same name. Was it intentional to hide that field with this variable?

Reply:
Yes, it made more sense to me to have the method return the value rather than modifying the field.